### DISCUSSION TITLE
Auto Validator

### DISCUSSION THREAD

**Discussion by NourEldin-Ali** [eclipse-glsp/glsp#1264] [Q&A]
Hi,

I'm trying to add auto validator to my model without click a button. This validator can be after adding a speficific element or automatically each X seconds.

Could you please help me to know how I can call [RequestMarkersAction](https://eclipse.dev/glsp/documentation/protocol/#210-element-validation) without click on the validation button of the palette?

![image](https://github.com/eclipse-glsp/glsp/assets/61513661/fe06700d-34b0-4275-8a7f-07d2e8bd8f59)

Thank you!

**Comment by planger** [comment]
Hi,

In GLSP 2.x, there is now explicit support for live validation, which matches your requirements, I believe. See https://github.com/eclipse-glsp/glsp/issues/980.

Essentially, it means that after each model change, the validation is triggered and potentially resulting problem markers are sent to the client. In order to use it, you'd just have to implement the `doLiveValidation()` method on the server (see [PR](https://github.com/eclipse-glsp/glsp-server/pull/200/files#diff-2af29cc519aa4fbfeed9d9c3586c4def0e6153e45be1ef000cad2329ce74d018)).

Hope this helps!

**Reply by NourEldin-Ali** [reply]
Thank you for your reply.

I'm still working on GLSP 1.1.0 RC 3, Can I do live validation using this version?

**Reply by planger** [reply]
Yes, with a customization of the model submission handler, which would always trigger the validator, similar to how it has been done here: https://github.com/eclipse-glsp/glsp-server/pull/200/files#diff-f807069b93d8a6693185add1bdb01df95c6685ccdbfd83e6187a347269f3e45e
